Sources/Core/State/PlayingState.swift

```diff
@@ -128,34 +130,35 @@ final class PlayingState: PlayerState {
         guard let media = context.currentMedia
             else { assertionFailure("media should exist"); return }
 
-        itemPlaybackObservingService.onPlaybackStalled = { [weak self] in self?.redirectToWaitingForNetworkState()
+        itemPlaybackObservingService.onPlaybackStalled = { [weak self] in
+            self?.redirectToWaitingForNetworkState()
         }
         itemPlaybackObservingService.onFailedToPlayToEndTime = { [weak self] in self?.redirectToWaitingForNetworkState()
         }
-        itemPlaybackObservingService.onPlayToEndTime = { [weak self, context] in
-            context.delegate?.playerContext(didItemPlayToEndTime: context.currentTime)
-            context.plugins.forEach { $0.didItemPlayToEndTime(media: media, endTime: context.currentTime) }
-            if context.loopMode {
-                self?.seek(position: 0)
+        itemPlaybackObservingService.onPlayToEndTime = { [weak self] in
+            guard let strongSelf = self else { return }
```

Comment:
You can now write:
```swift 
guard let self = self else { return }
```
I believe it is nicer

Reply:
not for me ;)